xml/System.IO/FileSystemWatcher.xml

```diff
@@ -823,20 +823,46 @@
   
  The <xref:System.IO.FileSystemWatcher.WaitForChanged%2A> method allows event handlers to be invoked to respond to file changes even if this property is set to `false`.  
   
-   
-  
+ If an exception <xref:System.IO.IOException> is thrown
```

Comment:
In general we never add scripts to the docs. Moreover it does not solve the problem as it modifies the file that is not persisted between system reboots.

We should rather describe where the setting comes from and mention the file (`/etc/sysctl.conf`) and name (` fs.inotify.max_user_instances`) of the setting that needs to be modified.

cc @tmds

Reply:
As I already explained, persisting this value between system reboots does not solve the problem.

Reply:
"exception IOException" sounds redundant.
```suggestion
 If an <xref:System.IO.IOException> is thrown
```

Reply:
I agree that "mention a persistent solution + applying the configuration to the running kernel" is a more valuable hint to users that read this doc page.

Instead of changing `/proc/sys/fs/inotify/max_user_instances` can you please mention the solution where you change `/etc/sysctl.conf` and then apply the change with `sudo sysctl -p`?

Reply:
This is not a solution.  This parameter can be changed live at any time.

Reply:
Editing /etc/sysctl.conf as siggested [here](https://github.com/dotnet/aspnetcore/issues/8449#issuecomment-512275929) changes the parameter instantly + every time the OS is booted. 
Do you mind explaining why that is not a solution (or point me to where you explained it since you say already did) or why your suggested script is better?

Reply:
1. Editing `/etc/sysctl.conf` changes the parameter at the next reboot.  The kernel does not observe changes to this configuration file.
2. The latter command applies the change to the kernel but does not work on BSD systems, as discussed there.
3. The number 524288 is arbitrary and potentially harmful.  256 is enough for my Linux system.  But every system is different, so a more adaptive approach should be recommended, such as the one implemented in my script.

Reply:
> Tagging @tmds as he might know some ways to improve [this script](https://review.docs.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher.enableraisingevents?view=net-5.0&branch=pr-en-us-7627#remarks).

This is an open source project now.